### PR TITLE
Ported to python3

### PR DESCRIPTION
**Pull request by benhg**

### CODE REVIEW

**Comment by benhg** [comment]
These are failing because Travis is set to run with Python2.7, but obviously due to the nature of our changes (involving python3), it will fail. We're going to make travis run the integration tests on Py3.

**Comment by benhg** [comment]
I think this has been fixed by the wonderful @roshbhatia. The tests should all pass now.

File: scripts/.#uninstall.sh [hunk 1]
```diff
@@ -0,0 +1 @@
+toggle@osboxes.4290:1551218152
```
Comment (eliasbakken) [inline scripts/.#uninstall.sh]: Is this necessary?

Comment (ThatWileyGuy) [reply scripts/.#uninstall.sh]: And why are there three of them?
And why does the only one with a meaningful command in it delete the entire OS? :D

File: toggle/CascadingConfigParser.py [hunk 2]
```diff
@@ -18,15 +18,15 @@
...
 """
 
-import ConfigParser
+import configparser
 import os
 import logging
 
 
-class CascadingConfigParser(ConfigParser.SafeConfigParser):
+class CascadingConfigParser(configparser.SafeConfigParser):
```
Comment (ThatWileyGuy) [inline toggle/CascadingConfigParser.py:26]: Are you seeing any warnings that SafeConfigParser is deprecated?
I saw them when I started poking at 2to3 in Redeem, but if this works without warnings we can address it later?

File: toggle/jogtest.py [hunk 3]
```diff
@@ -0,0 +1,3 @@
+from Jog import Jog
+
+print("Hello")
```
Comment (ThatWileyGuy) [inline toggle/jogtest.py]: And what's up with this?

File: toggle/pkg_resources [hunk 4]
```diff
@@ -0,0 +1,610 @@
+# This file helps to compute a version number in source trees obtained from
```
Comment (ThatWileyGuy) [inline toggle/pkg_resources]: Does this need to be checked in or does Versioneer generate this on demand?

If this does need to be checked it, can it be checked in separately from all the 2to3 changes?

**Review by ThatWileyGuy — CHANGES_REQUESTED** [review]

**Comment by ThatWileyGuy** [comment]
In general, this is fine. There are a few mysterious changes that should be explained or removed, and I'll sign off.

It would also be great to squash these commits so you're not adding and later removing files - check out `git rebase -i HEAD~6` and asking git to squash all your commits into the first one. You'll have to force-push to update the PR, but it should make things a lot cleaner.

File: toggle/pkg_resources (hunk 4 above)
Comment (ThatWileyGuy) [reply toggle/pkg_resources]: It looks like you added and then removed Versioneer and this was left over?

**Comment by roshbhatia** [comment]
@ThatWileyGuy Ben and I finished + squashed the changes, can you please check to see if we did ok?

**Comment by ThatWileyGuy** [comment]
Looks good to me, and I think Github will automagically squash away that merge commit for you.

**Review by ThatWileyGuy — APPROVED** [review]

**Comment by Wackerbarth** [comment]
Did anyone actually try this on real hardware before the merge? I get a failure to start the toggle service.

**Comment by goeland86** [comment]
I was having difficulties on my BBBW to get the Clutter backend to load - but didn't get it working on HW yet. The fact that it works in the VM is already a good indication the issue lies in underlying libraries rather than in the code itself though.

**Comment by Wackerbarth** [comment]
Yes, but, until it actually works, I think that it needs to be kept as a "proposed feature" rather than being merged into the master "dev" branch.

**Comment by goeland86** [comment]
I would argue that we should repair it directly in the `dev` branch now if there is anything to fix, and Umikaze RC3 will ship with the py3 version of toggle. I'm trying to build it this weekend.

**Comment by Wackerbarth** [comment]
That is where we have a serious difference of opinion with respect to workflow. You seem to dump in things that "probably will work" and make everyone work around the failures. OTOH, I don't want to merge in anything until it has been demonstrated to work.

As to Umikaze RC3, I would argue that we should have released a python2 version of Toggle 1.3 and already created that candidate. Instead, you seem to be willing to continue to delay things even more.
At this rate, we will never actually have a release.

**Comment by ThatWileyGuy** [comment]
Sorry, for some reason I had it in my head that the students had actual hardware. I'll take a look at what's broken.

**Comment by goeland86** [comment]
They do have hardware - but they're developing in parallel tracks between
on-hardware and VM, to advance faster.

**Comment by ThatWileyGuy** [comment]
Ah, okay.

The only thing broken is that pyconnman is not compatible with Python 3. You have to comment out the relevant lines in get_manager() in toggle/Network.py so we don't try to import it.

Aside from that...
- delete and recreate the virtualenv with `python -m virtualenv -p python3 --system-site-packages /usr/local/Toggle-develop`
- `python setup.py develop`
- `apt install python3-gi python3-gi-cairo python3-numpy python3-tornado python3-networkmanager`

**Comment by ThatWileyGuy** [comment]
Looks like someone opened a PR to fix pyconnman, but the project has been dormant for 5 years and it's not clear if it'll get merged: https://github.com/liamw9534/pyconnman/pull/4

**Comment by goeland86** [comment]
In any case we don't have connman installed as part of Umikaze normally -
if we do we should actually remove it from the base image, as it often
conflicts with NetworkManager anyway.

**Pull request merged**